cmd/ci-secret-bootstrap/main_test.go

```diff
@@ -0,0 +1,1420 @@
+  name: default
+current-context: default
+kind: Config
+preferences: {}
+users:
+- name: system:serviceaccount:ci:tool/api-ci-openshift-org:443
+  user:
+    token: token1
+- name: system:serviceaccount:ci:tool/api-build01-ci-devcluster-openshift-com:6443
+  user:
+    token: token2
+`
+)
+
+var (
+	configDefault = rest.Config{
+		Host:        "https://api.ci.openshift.org:443",
+		BearerToken: "token1",
+	}
+	configBuild01 = rest.Config{
+		Host:        "https://api.build01.ci.devcluster.openshift.com:6443",
+		BearerToken: "token2",
+	}
+)
+
+func TestCompleteOptions(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Errorf("Failed to create temp dir")
+	}
+	defer func() {
+		if err := os.RemoveAll(dir); err != nil {
+			t.Errorf("Failed to remove temp dir")
+		}
+	}()
+
+	bwPasswordPath := filepath.Join(dir, "bwPasswordPath")
+	configPath := filepath.Join(dir, "configPath")
+	kubeConfigPath := filepath.Join(dir, "kubeConfigPath")
+
+	fileMap := map[string][]byte{
+		bwPasswordPath: []byte("topSecret"),
+		configPath:     []byte(configContent),
+		kubeConfigPath: []byte(kubeConfigContent),
+	}
+
+	for k, v := range fileMap {
+		if err := ioutil.WriteFile(k, v, 0755); err != nil {
+			t.Errorf("Failed to remove temp dir")
+		}
+	}
+
+	testCases := []struct {
+		name               string
+		given              options
+		expectedError      error
+		expectedBWPassword string
+		expectedConfig     config
+		expectedClusters   []string
+	}{
+		{
+			name: "basic case",
+			given: options{
+				bwUser:         "username",
+				bwPasswordPath: bwPasswordPath,
+				configPath:     configPath,
+				kubeConfigPath: kubeConfigPath,
+			},
+			expectedBWPassword: "topSecret",
+			expectedConfig: config{
+				secretConfig{
+					From: map[string]bitWardenContext{
+						"key-name-1": {
+							BWItem: "item-name-1",
+							Field:  "field-name-1",
+						},
+						"key-name-2": {
+							BWItem: "item-name-1",
+							Field:  "field-name-2",
+						},
+						"key-name-3": {
+							BWItem:     "item-name-1",
+							Attachment: "attachment-name-1",
+						},
+						"key-name-4": {
+							BWItem: "item-name-2",
+							Field:  "field-name-1",
+						},
+						"key-name-5": {
+							BWItem:     "item-name-2",
+							Attachment: "attachment-name-1",
+						},
+						"key-name-6": {
+							BWItem:     "item-name-3",
+							Attachment: "attachment-name-2",
+						},
+					},
+					To: []secretContext{
+						{
+							Cluster:   "default",
+							Namespace: "namespace-1",
+							Name:      "prod-secret-1",
+						},
+						{
+							Cluster:   "build01",
+							Namespace: "namespace-2",
+							Name:      "prod-secret-2",
+						},
+					},
+				},
+			},
+			expectedClusters: []string{"build01", "default"},
+		},
+		{
+			name: "production case",
+			given: options{
+				bwUser:         "username",
+				bwPasswordPath: bwPasswordPath,
+				configPath:     "config.yaml",
```

Comment:
This one checks the production `config.yaml`.

It will result in the fact that each time we change on the production config, we need to update the test.
I think it is a "not-too-bad" burden because it gives us a chance to double check the changes on correctness of production config.

Reply:
Please. no -- that sounds like a massive pain in the ass and will break people working on `ci-tools`

Reply:
Removed the case although I cannot see why it will break people working on `ci-tools`.